api/src/transactions.rs

```diff
@@ -209,6 +210,59 @@ impl TransactionsApi {
             .await
     }
 
+    /// Wait for transaction by hash
+    ///
+    /// Same as /transactions/by_hash, but will wait for the transaction to be committed. To be used as a long poll
+    /// on clients (the "long" poll will return on the order of several seconds, according to the server configuration).
+    /// The request will return immediately once the committed transaction is found.
+    /// If the committed transaction is not found within the timeout, a 404 will be returned.
+    /// In case of other errors, the request will return immediately with the error.
+    #[oai(
+        path = "/transactions/wait_by_hash/:txn_hash",
+        method = "get",
+        operation_id = "wait_transaction_by_hash",
+        tag = "ApiTags::Transactions"
+    )]
+    async fn wait_transaction_by_hash(
```

Comment:
Do you know if there is any limit on number of concurrent connection the server can support? A concern might be that we exhaust that limit if we have too many of this blocking call. Also, a similar issue might happen for the threadpool exhaustion - can we make this call in the context of a blocking threadpool?